### PR TITLE
port toranj test9 insecure_traffic_join

### PR DESCRIPTION
**Pull request by priyankaCh22**
result for this is PASS.
[silk_insecure_traffic.log](https://github.com/openthread/silk/files/5450069/silk_insecure_traffic.log)

### CODE REVIEW

File: silk/tests/openthread/ot_test_insecure_traffic_join.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        self.assertEqual(self.router.getprop(wpan.WPAN_XPANID), self.parent.getprop(wpan.WPAN_XPANID))
+        self.assertEqual(self.router.getprop(wpan.WPAN_PANID), self.parent.getprop(wpan.WPAN_PANID))
+
+    @testcase.test_method_decorator
+    def test03_transmit_receive_insecure_traffic(self):
+        # Send insecure traffic from router to parent using link-local IP address
+        # for src/dst and insecure port number
+        self.transmit_receive_udp_message(self.router, self.parent, self.router.ip6_lla, self.parent.ip6_lla,
+                                          src_port=SRC_PORT, port=INSECURE_PORT)
+
```
Comment (priyankaCh22) [inline silk/tests/openthread/ot_test_insecure_traffic_join.py:129]: Please check the SRC_PORT obtained here.

From toranj: for insecure traffic we need to find the receiver's random src port too.
i.e. "Get the random src port number used by node1" (https://github.com/openthread/openthread/blob/master/tests/toranj/test-009-insecure-traffic-join.py#L88) and then in (https://github.com/openthread/openthread/blob/master/tests/toranj/test-009-insecure-traffic-join.py#L93) the rx_port is associated with node2. This made rx_port a little confusing. 

Please find below the code snippet being discussed. Thanks

```
"Send insecure traffic from node2 to node1 using link-local IP address
for src/dst and insecure port number"

sender = node2.prepare_tx(ll2, (ll1, insecure_port), "Hi (insecure)", NUM_MSGS)
recver = node1.prepare_rx(sender)
wpan.Node.perform_async_tx_rx()
verify(sender.was_successful)
verify(recver.was_successful)


"Get the random src port number used by node1 and ensure node2 allows
insecure rx traffic on that port"

rx_port = recver.all_rx_msg[0][1][1]
node2.permit_join(duration_sec='100', port=str(rx_port))



"Send insecure reply from node1 to node2"

sender2 = node1.prepare_tx((ll1, insecure_port), (ll2, rx_port), "Hi back! (insecure)", NUM_MSGS)
recver2 = node2.prepare_rx(sender2)
wpan.Node.perform_async_tx_rx()
verify(sender2.was_successful)
verify(recver2.was_successful)
```

File: silk/node/wpantund_base.py [hunk 2]
```diff
@@ -476,19 +476,20 @@ def timed_ping6(self, ipv6_target, num_pings, payload_size=8, interface=None):
...
+    def send_udp_data(self, target: str, port: int, message: str, source: str = None,
+                      src_port: int = random.randint(11200, 11400), hop_limit: int = 64):
         """Perform netcat command to send UDP message to ipv6_target via port.
 
         Args:
             target (str): target address.
             port (int): target port.
             message (str): message to send.
             source (str, optional): source address. Defaults to None.
+            src_port (int, optional): source port. Defaults to None.
```
Comment (qilinkang) [inline silk/node/wpantund_base.py]: default to a random number between 11200 and 11400 ?

The change is a dup of PR#86 ?

**Review by qilinkang — APPROVED** [review]
LGTM

File: silk/tests/openthread/ot_test_insecure_traffic_join.py (hunk 1 above)
Comment (jwhui) [reply silk/tests/openthread/ot_test_insecure_traffic_join.py:129]: @abtink, can you help take a look?

Comment (abtink) [reply silk/tests/openthread/ot_test_insecure_traffic_join.py:129]: The flow of the test as follows:
- We have two nodes. 
- Node 1 has a network (attached) and node 2 attaches insecurely (not knowing the master key)
- On node 1 we allow specific `insecure_port` number (1234)
   - This means node1 is allowed to receive message where dst port is `insecure_port` (1234) (with any src port).
- We then send a message from node 2 to node 1, where we use `insecure_port` as dst port and a random port number as src port.
   - We expect and check that node 1 does indeed receive this ok.
- Now on node 1 we check what the src port of the received msg is (which is saved as `rx_port`).
- Then on node 2, we allow the `rx_port` (`node2.permit_join(duration_sec='100', port=str(rx_port))`
   - This again means, node 2 now can receive/accept insecure traffic where dst port is set to `rx_port` value
- Finally we send an insecure msg this time from node 1 to node 2 using `rx_port` as dst port.

What we care is to use two different port number (for node 1 and node 2). We can also consider using two different hard-coded port values (no need to use random one)

Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_insecure_traffic_join.py:129]: Thanks for the detailed explanation Abtin. This clarifies the doubt about ports. And I think the test logic used to define SRC_PORT in advance is doing the same thing.

**Review by jwhui — APPROVED** [review]

**Pull request merged**